### PR TITLE
[ready] switch api format to new structure

### PR DESCRIPTION
**Pull request by MoKob**
This PR aims at the implementation of https://github.com/Project-OSRM/osrm-backend/issues/2675. See the discussion over there for details.

### CODE REVIEW

File: docs/http.md [hunk 1]
```diff
@@ -566,10 +569,6 @@ A `Lane` represents a turn lane at the corresponding turn location.
 #### Example
 
 ```json
```
Comment (TheMarex) [inline docs/http.md]: This was probably removed by accident?

Comment (MoKob) [inline docs/http.md]: It is moved down to the intersections part, where the lanes now live.

Comment (TheMarex) [inline docs/http.md]: I think it is fine to have both. This would document the structure of the `Lane` object.

**Comment by MoKob** [comment]
```{  
   "distance":641.6,
   "duration":49.2,
   "name":"Bernauer Straße",
   "maneuver":{  
      "type":"turn",
      "modifier":"left",
      "bearing_before":155,
      "bearing_after":66,
      "location":[  
         13.395944,
         52.538241
      ]
   },
   "geometry":"_kt_IskwpAG]GWAKKe@UmAUuAe@eCG[s@wDIe@o@iDMm@?CMq@y@iEu@uDCQSeA]mBk@eDE]AECS",
   "mode":"driving",
   "intersections":[  
      {  
         "location":[  
            13.395944,
            52.538241
         ],
         "bearings":[  
            60,
            150,
            240,
            330
         ],
         "lanes":[  
            {  
               "indications":[  
                  "left"
               ],
               "valid":true
            },
            {  
               "valid":false,
               "indications":[  
                  "straight",
                  "right"
               ]
            }
         ],
         "entry":[  
            true,
            true,
            true,
            false
         ],
         "in":3,
         "out":0
      },
      {  
         "location":[  
            13.396086,
            52.53828
         ],
         "bearings":[  
            60,
            150,
            240,
            330
         ],
         "lanes":[  
            {  
               "valid":false,
               "indications":[  
                  "left"
               ]
            },
            {  
               "indications":[  
                  "straight"
               ],
               "valid":true
            },
            {  
               "valid":true,
               "indications":[  
                  "straight",
                  "right"
               ]
            }
         ],
         "entry":[  
            true,
            false,
            false,
            true
         ],
         "in":2,
         "out":0
      },
      {  
         "location":[  
            13.397954,
            52.538801
         ],
         "bearings":[  
            60,
            240,
            330
         ],
         "entry":[  
            true,
            false,
            true
         ],
         "in":1,
         "out":0
      },
      {  
         "location":[  
            13.399011,
            52.539099
         ],
         "bearings":[  
            60,
            150,
            240,
            330
         ],
         "entry":[  
            true,
            true,
            false,
            true
         ],
         "in":2,
         "out":0
      },
      {  
         "location":[  
            13.401557,
            52.539822
         ],
         "bearings":[  
            60,
            150,
            240,
            330
         ],
         "lanes":[  
            {  
               "valid":false,
               "indications":[  
                  "left"
               ]
            },
            {  
               "indications":[  
                  "straight"
               ],
               "valid":true
            },
            {  
               "valid":true,
               "indications":[  
                  "straight",
                  "right"
               ]
            }
         ],
         "entry":[  
            true,
            true,
            false,
            true
         ],
         "in":2,
         "out":0
      },
      {  
         "location":[  
            13.402469,
            52.540086
         ],
         "bearings":[  
            60,
            165,
            240
         ],
         "entry":[  
            true,
            true,
            false
         ],
         "in":2,
         "out":0
      },
      {  
         "location":[  
            13.402563,
            52.540113
         ],
         "bearings":[  
            60,
            240,
            330
         ],
         "entry":[  
            true,
            false,
            true
         ],
         "in":1,
         "out":0
      }
   ]
}```

Example Intersection. Seems to work out fine now.

**Comment by TheMarex** [comment]
Great merging this as soon as travis is :green_apple: and then we can do another RC.

**Pull request merged**